### PR TITLE
Pull properties to support extra data on Azure Files

### PR DESCRIPTION
**Pull request by adreed-msft**
This fixes the lack of LMT data or otherwise from Azure Files in `azcopy list`.

### CODE REVIEW

File: cmd/list.go [hunk 1]
```diff
@@ -237,7 +237,7 @@ func (cooked cookedListCmdArgs) HandleListContainerCommand() (err error) {
 		}
 	}
 
-	traverser, err := InitResourceTraverser(source, cooked.location, &ctx, &credentialInfo, common.ESymlinkHandlingType.Skip(), nil, true, false, false, common.EPermanentDeleteOption.None(), func(common.EntityType) {}, nil, false, common.ESyncHashType.None(), common.EPreservePermissionsOption.None(), pipeline.LogNone, common.CpkOptions{}, nil, false, cooked.trailingDot, nil)
+	traverser, err := InitResourceTraverser(source, cooked.location, &ctx, &credentialInfo, common.ESymlinkHandlingType.Skip(), nil, true, true, false, common.EPermanentDeleteOption.None(), func(common.EntityType) {}, nil, false, common.ESyncHashType.None(), common.EPreservePermissionsOption.None(), pipeline.LogNone, common.CpkOptions{}, nil, false, cooked.trailingDot, nil)
```
Comment (gapra-msft) [inline cmd/list.go:240]: Do we support testing list yet in the e2e test? would it be worth adding it now as part of this PR?

Comment (adreed-msft) [reply cmd/list.go:240]: No, we don't. It would be quite a bit of effort.

**Review by gapra-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
Can we add a test for this case?

**Comment by adreed-msft** [comment]
> Can we add a test for this case?

Adjustments need to be made to the e2e framework, but possibly yes.

**Comment by seanmcc-msft** [comment]
> > Can we add a test for this case?
> 
> Adjustments need to be made to the e2e framework, but possibly yes. AzCopy list is not currently tested in any of the frameworks.

What would the time investment be, and would the changes be useful in the future for testing other cases?

**Review by vibhansa-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

**Review by nakulkar-msft — APPROVED** [review]

**Review by siminsavani-msft — APPROVED** [review]

**Pull request merged**